### PR TITLE
Add integrity-check command to irmin-fsck

### LINKED ISSUE #1195: Add integrity-check to the irmin-pack cli
`integrity-check` (in [irmin_pack](https://github.com/mirage/irmin/blob/master/src/irmin-pack/store.ml#L307) and [irmin-pack_layers](https://github.com/mirage/irmin/blob/master/src/irmin-pack/irmin_pack_layers.ml#L563)) could also be part of the offline checks we propose in irmin-fsck.

### PR DESCRIPTION
**Pull request by zshipko**
- Adds an `integrity-check` command to `irmin-fsck`
- Updates `src/irmin-pack/checks_intf.ml` with `Integrity_check` and `Reconstruct_index` definitions
    * Needed for https://github.com/tarides/tezos/issues/8
- Closes #1195 
- Updates `Irmin_pack_layers.S.check_integrity` function return type

### CODE REVIEW

File: src/irmin-pack/checks.ml [hunk 1]
```diff
@@ -176,6 +176,34 @@ struct
...
+  module Integrity_check = struct
+    module Store = Ext.Make (Conf) (M) (C) (P) (B) (H) (Node) (Commit)
+
+    let conf root = Config.v ~readonly:false ~fresh:false root
+
+    let auto_repair =
+      let open Cmdliner.Arg in
+      value
+      & flag
+        @@ info ~doc:"Automatically repair issues" ~docv:"AUTOREPAIR"
```
Comment (CraigFe) [inline src/irmin-pack/checks.ml:188]:
```suggestion
        @@ info ~doc:"Automatically repair issues"
```

I believe this has no effect for flags. (c.f. https://erratique.ch/software/cmdliner/doc/Cmdliner.Arg.html)

**Review by CraigFe — APPROVED** [review]

**Comment by icristescu** [comment]
This only works for normal (i.e. not layered) stores. Could you detect if it's a layered store (maybe using https://github.com/mirage/irmin/blob/master/src/irmin-pack/checks.ml#L45 ?) and call the corresponding [integrity-check](https://github.com/mirage/irmin/blob/master/src/irmin-pack/irmin_pack_layers.ml#L563) if it is?

**Comment by zshipko** [comment]
@icristescu I made that update and think this is ready for me to merge if you everything looks good

**Review by icristescu — APPROVED** [review]
Thanks, this looks good! Let's wait for the CI and the you can merge it.

**Pull request merged**